### PR TITLE
Refactor code and add tests

### PR DESCRIPTION
**Pull request by markdboyd**
## Changes proposed in this pull request:

- This PR refactors the code extensively to add interfaces and structs for testing
- Remove the separate `sandbox` package which seemed unnecessary and made testing more difficult
- Make everything private by default
- Add unit test for purge functionality
- Update code so that recipients for notifications and purge are retrieved separately. Current code had a bug where the recipients from the last notified space would be using when purging spaces in the same org

## security considerations

None, just refactoring code for testability and maintenance

### CODE REVIEW

**Review by rcgottlieb — DISMISSED** [review]

File: cmd/purge/notify.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+func notifySpaceUsers(
+	ctx context.Context,
+	cfClient *cfResourceClient,
+	opts Options,
+	userGUIDs map[string]bool,
+	org *resource.Organization,
+	details SpaceDetails,
+	mailSender mailer,
+) error {
+	notifyTemplate, err := template.ParseFiles("../../templates/base.html", "../../templates/notify.tmpl")
```
Comment (jameshochadel) [inline cmd/purge/notify.go:22]: At some point I think you'd enjoy refactoring this to use `embed` — not necessary now, but maybe a future Friday project

File: cmd/purge/notify.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		return fmt.Errorf("error listing users on space %s: %w", details.Space.Name, err)
+	}
+
+	recipients, err := listRecipients(userGUIDs, spaceUsers)
+	if err != nil {
+		return fmt.Errorf("error listing recipients on space %s: %w", details.Space.Name, err)
+	}
+
+	log.Printf("Notifying space %s; recipients %+v", details.Space.Name, recipients)
+	if !opts.DryRun {
```
Comment (jameshochadel) [inline cmd/purge/notify.go:38]: How about `if opts.DryRun { return nil }`, and then deindent this section? Similar to the error handling code

Comment (jameshochadel) [reply cmd/purge/notify.go:38]: I could go either way on this — let me know what you think

File: cmd/purge/purge.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+	recipients, err := listRecipients(userGUIDs, spaceUsers)
+	if err != nil {
+		return fmt.Errorf("error listing recipients on space %s: %w", details.Space.Name, err)
+	}
+
+	developers, managers := listSpaceDevsAndManagers(userGUIDs, spaceRoles)
+	log.Printf("Purging space %s; recipients: %+v, developers: %+v, managers: %+v", details.Space.Name, recipients, developers, managers)
+
+	if !opts.DryRun {
```
Comment (jameshochadel) [inline cmd/purge/purge.go:45]: If you want to change the dry run indentation, here's another — otherwise, you can just resolve this

**Review by jameshochadel — APPROVED** [review]
Looks great. Sensible abstractions, good names, clear control flow. It was easy to understand for someone not familiar with the codebase. One or two comments inline for your consideration but no showstoppers. Thanks Mark!

**Review by jameshochadel — APPROVED** [review]

**Pull request merged**